pkg/cmd/dcos_auth_listproviders.go

```diff
@@ -0,0 +1,120 @@
+package cmd
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/dcos/dcos-cli/pkg/httpclient"
+	"github.com/olekukonko/tablewriter"
+	"github.com/spf13/cobra"
+	"os"
+)
+
+// These are the different auth types that DC/OS supports with the names that they'll be given from the providers
+// endpoint.
+const (
+	AuthTypeDCOSUidPassword = "dcos-uid-password"
+	AuthTypeDCOSUidServiceKey = "dcos-uid-servicekey"
+	AuthTypeDCOSUidPasswordLDAP = "dcos-uid-password-ldap"
+	AuthTypeSAMLSpInitiated = "saml-sp-initiated"
+	AuthTypeOIDCAuthCodeFlow = "oidc-authorization-code-flow"
+	AuthTypeOIDCImplicitFlow = "oidc-implicit-flow"
+)
+
+var jsonOutput bool
+
+// authCmd represents the `dcos auth` subcommand.
+var authListProvidersCmd = &cobra.Command{
+	Use: "list-providers",
+	Run: listProviders,
+}
+
+func init() {
+	authCmd.AddCommand(authListProvidersCmd)
+	authListProvidersCmd.Flags().BoolVar(&jsonOutput, "json", false,
+		"returns providers in json format")
```

Comment:
There is no line-length limit in Go, we use ~110 characters as a rule of thumb, a single line is fine here.